pkg/util/validation/exporter/ring.go

```diff
@@ -114,16 +123,41 @@ func (c *RingConfig) toRingConfig() ring.Config {
 }
 
 // overridesExporterRing is a ring client that overrides-exporters can use to
-// assume ownership of a tenant shard
+// establish a leader replica that is the unique exporter of per-tenant limit metrics.
 type overridesExporterRing struct {
-	config     RingConfig
+	services.Service
+
+	config RingConfig
+
 	client     *ring.Ring
 	lifecycler *ring.BasicLifecycler
+
+	subserviceManager *services.Manager
+	subserviceWatcher *services.FailureWatcher
+	logger            log.Logger
 }
 
-// Owns looks up whether this ring member owns the given tenant.
-func (o *overridesExporterRing) Owns(tenantID string) (bool, error) {
-	return instanceOwnsIdentifier(o.client, o.lifecycler.GetInstanceAddr(), tenantID)
+// isLeader checks whether this instance is the leader replica that exports metrics for all tenants.
+func (r *overridesExporterRing) isLeader(at time.Time) (bool, error) {
+	// Instances in state ring.LEAVING are not eligible to be the leader.
+	if r.lifecycler.GetState() == ring.LEAVING {
```

Comment:
I would remove this check. Don't think adds much value. A LEAVING instance will have its code running for a very short time, cause then will terminate.

Reply:
Taking this out makes the ring unit test in its current form fail, specifically this part.
```
// Since the previous leader is still in the ring but in state ring.LEAVING, there should be no leader now.
	require.True(t, !i1IsLeader && !i2IsLeader)
```
If you don't feel strongly about it I'd prefer leaving it in, otherwise I can take it out and rewrite that part of the test.